src/Endatix.Core/Entities/FormDefinition.cs

```diff
@@ -1,26 +1,43 @@
-using System.Collections.Generic;
 using Endatix.Core.Configuration;
 using Endatix.Core.Infrastructure.Domain;
 
 namespace Endatix.Core.Entities;
 
 public partial class FormDefinition : BaseEntity, IAggregateRoot
 {
-
-    public bool IsDraft { get; set; }
-    public int Version { get; set; }
-    public string JsonData { get; internal set; }
-    public long FormId { get; set; }
-    public Form Form { get; set; }
-    public bool IsActive { get; set; }
     private readonly List<Submission> _submissions = [];
-    public IReadOnlyCollection<Submission> Submissions => _submissions.AsReadOnly();
 
-    public FormDefinition(bool isDraft = false, string jsonData = null, bool isActive = true)
+    private FormDefinition() { }
+
+    public FormDefinition(bool isDraft = false, string? jsonData = null)
     {
         jsonData ??= EndatixConfig.Configuration.DefaultFormDefinitionJson;
         IsDraft = isDraft;
         JsonData = jsonData;
-        IsActive = isActive;
     }
-}
\ No newline at end of file
+
+    public IReadOnlyList<Submission> Submissions => _submissions.AsReadOnly();
+
+    public bool IsDraft { get; private set; }
+    public string JsonData { get; private set; } = EndatixConfig.Configuration.DefaultFormDefinitionJson;
+
+    public long FormId { get; private set; }
+
+    /// <summary>
+    /// Updates the form definition's JSON data with the provided value, or keeps the current data if null is provided.
+    /// </summary>
+    /// <param name="jsonData">The new JSON data for the form definition, or null to keep the current data.</param>
+    public void UpdateSchema(string? jsonData)
```

Comment:
Similarly to what I mention in the comment for UpdateToken in Submission - a method is added that simply sets the value of a property. It updates the json definition but is called UpdateSchema.
I understand this because I know the code base well but for someone not familiar, it could bring confusion.

Reply:
Yes, the properties are actually methods and their setters can have logic when we need any for e.g. validations or other stuff. Here we introduce a new term "schema", which we don't have anywhere in the context of the definition. This can be confusing.
I don't say we need to make changes now but still I mention what I see and think.